ravenframework/SupervisedLearning/FeatureSelection/RFE.py

```diff
@@ -0,0 +1,815 @@
+                               parametersToInclude: list of parameters to include
+                               whichSpace: which space? feature or target?
+
+      @ Out, score, float, the score for this combination
+      @ Out, survivors,list(str), the list of parameters for this combination
+      @ Out, featureCombination, numpy.ndarray(int), list of feature that should be tested (indeces of features)
+    """
+    # copy arrays that will be modified during the search
+    support_ = copy.copy(supportData['originalSupport'])
+    featuresForRanking = copy.copy(supportData['featuresForRanking'])
+    ranking_ = copy.copy(supportData['ranking_'])
+    supportOfSupport_ = copy.copy(supportData['supportOfSupport_'])
+    # get immutable settings
+    nFeaturesToSelect = supportData['nFeaturesToSelect']
+    subGroups = supportData['subGroups']
+    step = supportData['firstStep']
+    setStep = supportData['setStep']
+    mask = supportData['mask']
+    nFeatures = supportData['nFeatures']
+    nTargets = supportData['nTargets']
+    nParams = supportData['nParams']
+    targetsIds = supportData['targetsIds']
+    originalParams = supportData['originalParams']
+
+    parametersToInclude = supportData['parametersToInclude']
+    whichSpace = supportData['whichSpace']
+
+    # initialize working dir
+    indexToKeep = None
+    # the search is done at least once
+    doAtLeastOnce = True
+    while np.sum(support_) > nFeaturesToSelect or doAtLeastOnce:
+      # Remaining features
+      estimator = copy.deepcopy(estimatorObj)
+      raminingFeatures = int(np.sum(support_))
+      featuresForRanking = np.arange(nParams)[support_]
+      # subgrouping
+      outputToRemove = None
+      if outputspace != None:
+        indexToRemove = []
+        outputToRemove = []
+        indexToKeep = []
+        sg = copy.deepcopy(subGroups)
+        outputToKeep = sg.pop(groupId)
+        for grouptoremove in sg:
+          outputToRemove += grouptoremove
+        for t in outputToRemove:
+          indexToRemove.append(targetsIds.index(t))
+        for t in outputToKeep:
+          indexToKeep.append(targetsIds.index(t))
+        if whichSpace != 'feature':
+          supportOfSupport_[np.asarray(indexToRemove)] = False
+          supportOfSupport_[np.asarray(indexToKeep)] = True
+
+      supportIndex = 0
+      for idx in range(len(supportOfSupport_)):
+        if mask[idx]:
+          supportOfSupport_[idx] = support_[supportIndex]
+          supportIndex=supportIndex+1
+      if whichSpace == 'feature':
+        features = np.arange(nFeatures)[supportOfSupport_]
+        targets = np.arange(nTargets)
+      else:
+        features = np.arange(nFeatures)
+        targets = np.arange(nTargets)[supportOfSupport_]
+      # Rank the remaining features
+      print("(            ) FEATURE SELECTION - RFE  : Message         -> Fitting estimator with %d features." % np.sum(support_))
```

Comment:
change print to raiseAMessage?

Reply:
Unfortunately this can go in parallel. If I change it to raiseAMessage I need to pickle the message handler and send it to the remote workers.